homeassistant/components/zha/core/gateway.py

```diff
@@ -69,8 +86,15 @@ def device_removed(self, device):
         """Handle device being removed from the network."""
         for device_entity in self._device_registry[device.ieee]:
             self._hass.async_create_task(device_entity.async_remove())
```

Comment:
I'll remove this once I rewrite the API. See the note below. I have a plan for it but it will require UI work. I can do it as part of this if you prefer... it will just make this PR even larger...

Reply:
It's ok to fix in a later PR. :+1:

Reply:
Ok, I'll see if I can figure out how to handle via signals

Reply:
> Config entry platforms remove entities by default when the corresponding entry is unloaded and unloading is forwarded to the platform domains.

that's all or nothing when removing ZHA itself no? I'm talking about removing a single sensor from the network... Just trying to make sure we're on the same page. :)

Reply:
Yes, I just mentioned it when talking about removing entities.

Just create a signal string, built from the entity_id, that we connect to the `async_remove` entity method, in `async_added_to_hass`.

Store the entity_ids of the added entities to know what entities to remove when the device leaves the network.

Reply:
Was just making sure. Thx! I’ll do this after lunch.

Reply:
Done